app/src/main/java/org/wikipedia/edit/EditAbuseFilterResult.kt

```diff
@@ -0,0 +1,31 @@
+package org.wikipedia.edit
+
+internal class EditAbuseFilterResult(val code: String?, val info: String?, val warning: String?) : EditResult("Failure") {
+
+    // to simply treat these as an error.
+    // This case is here because, unfortunately, an admin can create an abuse filter which
+    // emits an arbitrary error code over the API.
+    // TODO: More properly handle the case where the AbuseFilter throws an arbitrary error.
+    // Oh, and, you know, also fix the AbuseFilter API to not throw arbitrary error codes.
+    val type: Int
+        get() = if (code != null && code.startsWith("abusefilter-warning")) {
+            TYPE_WARNING
+        } else if (code != null && code.startsWith("abusefilter-disallowed")) {
+            TYPE_ERROR
+        } else if (info != null && info.startsWith("Hit AbuseFilter")) {
```

Comment:
Would it be good if we can use `code.orEmpty().startWith()` on these null check type of if statement?

Reply:
sure